packages/@n8n/benchmark/src/test-execution/app-metrics-poller.ts

```diff
@@ -0,0 +1,143 @@
+/**
+ * Polls the /metrics endpoint from an n8n instance to collect application metrics
+ * during benchmark test runs.
+ */
+export class AppMetricsPoller {
+	private intervalId: NodeJS.Timeout | undefined = undefined;
+	private metricsData: string[] = [];
+	private isRunning = false;
+
+	constructor(
+		private readonly metricsUrl: string,
+		private readonly pollIntervalMs: number = 5000,
+	) {}
+
+	/**
+	 * Starts polling the metrics endpoint
+	 */
+	start() {
+		if (this.isRunning) {
+			console.warn('Metrics poller is already running');
+			return;
+		}
+
+		// In case the old poller is somehow still running, make sure we stop it.
+		if (this.intervalId) {
+			clearInterval(this.intervalId);
+			this.intervalId = undefined;
+		}
+
+		this.isRunning = true;
+		this.metricsData = [];
+
+		// Immediately poll once to get initial metrics
+		void this.pollMetrics();
+
+		// Set up interval polling
+		this.intervalId = setInterval(() => {
+			void this.pollMetrics();
+		}, this.pollIntervalMs);
+	}
+
+	/**
+	 * Stops polling the metrics endpoint
+	 */
+	stop() {
+		if (this.intervalId) {
+			clearInterval(this.intervalId);
+			this.intervalId = undefined;
+		}
+		this.isRunning = false;
+	}
+
+	/**
+	 * Gets all collected metrics data
+	 */
+	getMetricsData(): string[] {
+		return this.metricsData;
+	}
+
+	/**
+	 * Polls the metrics endpoint once
+	 */
+	private async pollMetrics() {
+		try {
+			const response = await fetch(this.metricsUrl);
+
+			if (!response.ok) {
+				console.warn(`Failed to poll metrics: ${response.status} ${response.statusText}`);
+				return;
+			}
+
+			const metricsText = await response.text();
+			this.metricsData.push(metricsText);
+		} catch (error) {
+			console.warn(
+				`Error polling metrics: ${error instanceof Error ? error.message : String(error)}`,
+			);
+		}
+	}
+}
+
+/**
+ * Parses Prometheus metrics text format and extracts time series data
+ */
+export class PrometheusMetricsParser {
```

Comment:
nit: Could we separate these into separate files? Class per file is common pattern and makes them more discoverable